frontend/src/pages/offers/CreateOfferForm.tsx

```diff
@@ -0,0 +1,124 @@
+import { FunctionComponent, ReactNode, useContext, useMemo } from 'react'
+import { useForm } from 'react-hook-form'
+import { useParams } from 'react-router-dom'
+import Button from '../../components/Button'
+import ReadOnlyField from '../../components/forms/ReadOnlyField'
+import TextField from '../../components/forms/TextField'
+import { UserProfileContext } from '../../components/UserProfileContext'
+import {
+  GroupType,
+  OfferCreateInput,
+  useAllGroupsMinimalQuery,
+  useAllShipmentsQuery,
+} from '../../types/api-types'
+import { formatShipmentName } from '../../utils/format'
+
+interface Props {
+  /**
+   * If true, the Submit button will be disabled
+   */
+  isLoading: boolean
+  /**
+   * The text to display in the Submit button of the form
+   */
+  submitButtonLabel: ReactNode
+  /**
+   * The callback triggered when the user submits the form
+   */
+  onSubmit: (input: OfferCreateInput) => void
+}
+
+/**
+ * This form allows users to create a new offer. It is very specific to offer
+ * creation, and therefore difficult to reuse for editing offers.
+ */
+const OfferForm: FunctionComponent<Props> = (props) => {
+  // Each group leader is assigned to a single group
+  // We need to figure out which group that is
+  // TODO: handle the fact that admins can be assigned to any number of groups
+  const profile = useContext(UserProfileContext)
+  const { data: groups } = useAllGroupsMinimalQuery()
+
+  const groupForUser = useMemo(() => {
+    if (groups?.listGroups && profile) {
+      return groups.listGroups.find(
+        (group) =>
+          group.captainId === profile.id &&
+          group.groupType === GroupType.SendingGroup,
+      )
+    }
+  }, [groups, profile])
```

Comment:
Here we start to get into the weeds because we need data from multiple endpoints.

We want to display some information about the group the user is a captain of. To do that, we fetch the list of all the groups and figure out which one has a matching `captainId`. It would be better if we could query a group by captain ID in the future.

Reply:
How about an endpoint that's like `listEditableGroups()` that returns all the groups the user has edit access to? This would handle the admin and non-admin cases with a single endpoint.

Reply:
In that case I guess we'd need to give the option of selecting the sending group for users with multiple 🤔

Reply:
And on top of that, if we move to multiple sending/receiving groups per shipment, we'll need to filter out the groups that aren't part of that shipment. So maybe something like:

```ts
const groupsFromShipment = shipment.sendingGroups.map(group => group.id)
const groupsToPickFrom = groups.listGroups.filter(
  group => 
    group.captainId === profile.id &&
    group.groupType = GroupType.SendingGroup &&
    groupsFromShipment.includes(group.id)
)
```

And then on the UI we can display a dropdown if there are 2 or more choices.